hbase-backup/src/test/java/org/apache/hadoop/hbase/backup/TestIncrementalBackup.java

```diff
@@ -100,142 +141,227 @@ public void TestIncBackupRestore() throws Exception {
+    TEST_UTIL.getAdmin().modifyTable(newTable1Desc);
+
     try (Connection conn = ConnectionFactory.createConnection(conf1)) {
-      int NB_ROWS_FAM3 = 6;
-      insertIntoTable(conn, table1, fam3Name, 3, NB_ROWS_FAM3).close();
-      insertIntoTable(conn, table1, mobName, 3, NB_ROWS_FAM3).close();
-      Admin admin = conn.getAdmin();
-      BackupAdminImpl client = new BackupAdminImpl(conn);
+      BackupAdminImpl backupAdmin = new BackupAdminImpl(conn);
       BackupRequest request = createBackupRequest(BackupType.FULL, tables, BACKUP_ROOT_DIR);
-      String backupIdFull = takeFullBackup(tables, client);
-      validateRootPathCanBeOverridden(BACKUP_ROOT_DIR, backupIdFull);
-      assertTrue(checkSucceeded(backupIdFull));
-
-      // #2 - insert some data to table
-      Table t1 = insertIntoTable(conn, table1, famName, 1, ADD_ROWS);
-      LOG.debug("writing " + ADD_ROWS + " rows to " + table1);
-      Assert.assertEquals(HBaseTestingUtil.countRows(t1),
-        NB_ROWS_IN_BATCH + ADD_ROWS + NB_ROWS_FAM3);
-      LOG.debug("written " + ADD_ROWS + " rows to " + table1);
-      // additionally, insert rows to MOB cf
-      int NB_ROWS_MOB = 111;
-      insertIntoTable(conn, table1, mobName, 3, NB_ROWS_MOB);
-      LOG.debug("written " + NB_ROWS_MOB + " rows to " + table1 + " to Mob enabled CF");
-      t1.close();
-      Assert.assertEquals(HBaseTestingUtil.countRows(t1),
-        NB_ROWS_IN_BATCH + ADD_ROWS + NB_ROWS_MOB);
-      Table t2 = conn.getTable(table2);
-      Put p2;
-      for (int i = 0; i < 5; i++) {
-        p2 = new Put(Bytes.toBytes("row-t2" + i));
-        p2.addColumn(famName, qualName, Bytes.toBytes("val" + i));
-        t2.put(p2);
-      }
-      Assert.assertEquals(NB_ROWS_IN_BATCH + 5, HBaseTestingUtil.countRows(t2));
-      t2.close();
-      LOG.debug("written " + 5 + " rows to " + table2);
-      // split table1
-      SingleProcessHBaseCluster cluster = TEST_UTIL.getHBaseCluster();
-      List<HRegion> regions = cluster.getRegions(table1);
-      byte[] name = regions.get(0).getRegionInfo().getRegionName();
-      long startSplitTime = EnvironmentEdgeManager.currentTime();
-      try {
+      String fullBackupId = backupAdmin.backupTables(request);
+      assertTrue(checkSucceeded(fullBackupId));
+
+      TableName[] fromTables = new TableName[] { table1 };
+      TableName[] toTables = new TableName[] { table1_restore };
+      backupAdmin.restore(BackupUtils.createRestoreRequest(BACKUP_ROOT_DIR, fullBackupId, false,
+        fromTables, toTables, true, true));
+
+      Table table = conn.getTable(table1_restore);
+      Assert.assertEquals(NB_ROWS_IN_BATCH, HBaseTestingUtil.countRows(table));
+
+      int ROWS_TO_ADD = 1_000;
+      // different IDs so that rows don't overlap
+      insertIntoTable(conn, table1, fam1, 3, ROWS_TO_ADD);
+      insertIntoTable(conn, table1, mobFam, 4, ROWS_TO_ADD);
+
+      Admin admin = conn.getAdmin();
```

Comment:
When using the `Admin` accessed from the `Connection` instance, you need to close it. When using the `Admin` from the testing utility, you need to not close it.

I know.

Reply:
I do close the admin [here](https://github.com/apache/hbase/pull/6370/files#diff-21c088f224d8731ed61a0c97a72f528097ca02977cbc3195563fc996b1b66c38R364)

Reply:
I'll use a try/catch instead of closing it manually to make it more obvious